.github/workflows/deploy-standalone-plugins.yml

```diff
@@ -1,99 +1,154 @@
+            SLUG=${{ inputs.plugin }}
+            if echo $PLUGINS | jq -e '.[] | select(. == "'$SLUG'")' > /dev/null; then
+              PLUGINS="[ \"$SLUG\" ]"
             else
-              echo "The ${{ inputs.slug }} module slug is missing in the file plugins.json."
+              echo "::error::The plugin $SLUG is not in the list of plugins to deploy."
               exit 1
             fi
-          else
-            # Load the JSON file and parse from "{name: {slug, version}, ...}" to "include: [{ name, slug, version }, ...]"
-            # for use in the matrix.
-            # For plugins, the "version" parameter is not included here; it will dynamically get it in its own job.
-
-            # The "dry-run" parameter is included here to set the deployment mode.
-            # When running the manual (workflow_dispatch) workflow, this value will be set from manual input type.
-            echo "matrix=$(jq -c '{include: ([.modules | to_entries[] | {name:.key, slug: .value.slug, version: .value.version, directory: "build", "dry-run": false}] + ([.plugins[] | {name:. , slug:. , directory: "plugins", "dry-run": false}]))}' plugins.json)" >> $GITHUB_OUTPUT
           fi
+
+          # Set the matrix.
+          echo "::notice::Deploying the following plugins: $PLUGINS"
+          echo "plugins=$(echo $PLUGINS | jq -c .)" >> $GITHUB_OUTPUT
+
   deploy:
-    name: Deploy Plugin
-    needs: release
+    name: "Deploy Plugin: ${{ matrix.plugin }}"
+    needs: pre-run
     runs-on: ubuntu-latest
+    permissions:
+      actions: write
+      deployments: write
     strategy:
-      matrix: ${{ fromJSON(needs.release.outputs.matrix) }}
+      matrix:
+        plugin: ${{ fromJSON(needs.pre-run.outputs.matrix) }}
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v3
+      - name: Checkout
+        uses: actions/checkout@v4
+
       - name: Setup Node.js (.nvmrc)
-        uses: actions/setup-node@v3
+        uses: actions/setup-node@v4
         with:
           node-version-file: '.nvmrc'
           cache: npm
+
       - name: Install npm dependencies
         run: npm ci
-      - name: Building standalone plugins
-        if: ${{ matrix.directory != 'plugins' }}
-        run: npm run build-plugins
-      - name: Set version
-        id: set_version
+
+      - name: Build plugin
+        run: npm run build:plugin:${{ matrix.plugin }}
+
+      - name: Set plugin version
+        working-directory: ./build
         run: |
-          if [ -z "${{ matrix.version }}" ]; then
-            echo "version=$(node ./bin/plugin/cli.js get-plugin-version --slug=${{ matrix.slug }})" >> $GITHUB_OUTPUT
-          else
-            echo "version=${{ matrix.version }})" >> $GITHUB_OUTPUT
-          fi
-      - name: Deploy Standalone Plugin - ${{ matrix.slug }}
+          echo "PLUGIN_VERSION=$(jq -r '."${{ matrix.plugin }}"' manifest.json)" >> $GITHUB_ENV
```

Comment:
Curious: Why use `$GITHUB_ENV` instead of `$GITHUB_OUTPUT` like before?

Reply:
I think it's best to use `$GITHUB_OUTPUT` when you want to share the info across the jobs. For the same job, an env variable is suffice.